sdk/digitaltwins/Azure.DigitalTwins.Core/tests/ImportTests.cs

```diff
@@ -0,0 +1,163 @@
+{
+    /// <summary>
+    /// Tests for DigitalTwinServiceClient methods dealing with Digital Twin Import operations.
+    /// </summary>
+    public class ImportTests : E2eTestBase
+    {
+        public ImportTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task Import_Lifecycle()
+        {
+            // arrange
+            DigitalTwinsClient client = GetClient();
+
+            string jobId = await GetUniqueJobIdAsync(client, TestAssetDefaults.ImportJobId).ConfigureAwait(false);
+            var inputBlobUri = TestEnvironment.InputBlobUri;
+            var outputBlobUri = $"{TestEnvironment.StorageContainerEndpoint}/output-{jobId}.ndjson";
+            var importJob = new ImportJob(inputBlobUri, outputBlobUri);
+
+            try
+            {
+                // act + assert
+
+                // validate CREATE job
+                var createResponse = await client.CreateImportJobsAsync(jobId, importJob).ConfigureAwait(false);
+                Assert.IsNotNull(createResponse);
+                var rawCreateResponse = createResponse.GetRawResponse();
+                Assert.AreEqual((int)HttpStatusCode.Created, rawCreateResponse.Status);
+
+                // Validate GET job
+                var getResponse = await client.GetImportJobsByIdAsync(jobId).ConfigureAwait(false);
+                Assert.IsNotNull(getResponse);
+                var rawGetResponse = getResponse.GetRawResponse();
+                Assert.AreEqual((int)HttpStatusCode.OK, rawGetResponse.Status);
+
+                // validate LIST all jobs
+                var listResponse = await client.ListImportJobsAsync();
+                Assert.IsNotNull(listResponse);
+                var rawListResponse = listResponse.GetRawResponse();
+                Assert.AreEqual((int)HttpStatusCode.OK, rawListResponse.Status);
+
+                // validate CANCEL job
+                var cancelResponse = await client.CancelImportJobsAsync(jobId).ConfigureAwait(false);
+                Assert.IsNotNull(cancelResponse);
+                var rawCancelResponse = cancelResponse.GetRawResponse();
+                Assert.AreEqual((int)HttpStatusCode.OK, rawCancelResponse.Status);
+
+                // validate DELETE job
+                var deleteResponse = await client.DeleteImportJobsAsync(jobId).ConfigureAwait(false);
+                Assert.IsNotNull(deleteResponse);
+                Assert.AreEqual((int)HttpStatusCode.NoContent, deleteResponse.Status);
+
+                // validate GET job after deletion - should fail
+                Func<Task> act = async () => await client.GetImportJobsByIdAsync(jobId).ConfigureAwait(false);
+                act.Should().Throw<RequestFailedException>()
+                    .And.Status.Should().Be((int)HttpStatusCode.NotFound);
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail($"Failure in executing a step in the test case: {ex.Message}.");
```

Comment:
`ex.Message` can lose important context like call stack. Can we just serialize the entire exception or otherwise ensure it is available to allow failures to be easily triaged and/or root caused.

Reply:
Updated